packages/grid/x-grid-modules/src/hooks/features/useColumnReorder.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from 'react';
+import { ColDef } from '../../models/colDef';
+import { useLogger } from '../utils';
+import { ApiRef } from '../../models';
+
+export const useColumnReorder = (
+  apiRef: ApiRef
+) => {
+  const logger = useLogger('useColumnReorder');
+
+  const dragCol = React.useRef<ColDef | null>();
+  const dragColNode = React.useRef<HTMLElement | null>();
+
+  const handleDragStart = React.useCallback(
+    (col: ColDef, htmlEl: any): void => {
+      logger.debug(`Start dragging col ${col.field}`);
+      dragCol.current = col;
+      dragColNode.current = htmlEl;
+      dragColNode.current?.addEventListener('dragend', handleDragEnd);
```

Comment:
memory leak here? Or where do you clean the subscription?

Reply:
You can add `{once: true}` as 3rd parameter?

Reply:
I will add the 3rd param. In terms of where this is cleared -> it's inside the `handleDragEnd `

Reply:
What's our browser support policy?
https://material-ui.com/getting-started/supported-platforms/#browser
https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener

Reply:
Hmm ok but in the case of IE, the third param will just not work, be ignored I guess right? I have Virtual Box, I'll try and test it to make sure this throwing errors.

Reply:
I mean, the browser policy is unknown on the -x repository. it was meant as a question for @dtassone :)